### PR TITLE
Command code for adding pki-intermediate

### PR DESCRIPTION
**Pull request by divyapola5**
Command code to add intermediate CA

### CODE REVIEW

File: command/pki_add_intermediate.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	flagCommonName string
+}
+
+func (c *PKIAddIntermediateCommand) Synopsis() string {
+	return "Generate intermediate certificate"
+}
+
+func (c *PKIAddIntermediateCommand) Help() string {
+	helpText := `
+Usage: vault pki add-intermediate [options] [ROOT_MOUNT] PATH COMMON_NAME [K=V]
```
Comment (cipherboy) [inline command/pki_add_intermediate.go:30]: root mount is an option I think from the flags below? :)

File: command/pki_add_intermediate.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	return strings.TrimSpace(helpText)
+}
+
+func (c *PKIAddIntermediateCommand) Flags() *FlagSets {
+	set := c.flagSet(FlagSetHTTP)
+	f := set.NewFlagSet("Command Options")
+
+	f.StringVar(&StringVar{
+		Name:    "mount",
+		Target:  &c.flagMount,
```
Comment (cipherboy) [inline command/pki_add_intermediate.go:52]: I don't think this one is necessary? That's what the first positional argument should be, correct? `args[1]` in particular.

File: command/pki_add_intermediate.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	f.StringVar(&StringVar{
+		Name:    "root-mount",
+		Target:  &c.flagRootMount,
+		Default: "pki",
+		EnvVar:  "",
+		Usage:   "PKI root mount",
+	})
+
+	f.StringVar(&StringVar{
+		Name:    "common_name",
```
Comment (cipherboy) [inline command/pki_add_intermediate.go:67]: Same as above `flagMount` above. Only `root-mount` is necessary.

File: command/pki_add_intermediate.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+		return 1
+	}
+
+	args = f.Args()
+	if len(args) < 3 {
+		c.UI.Error(fmt.Sprintf("Not enough arguments (expected 3+, got %d)", len(args)))
+		return 1
+	}
+
+	rootMountPath := sanitizePath(args[0]) // TODO figure out how to check for optional fields
```
Comment (cipherboy) [inline command/pki_add_intermediate.go:101]: Ah I see now. You can remove this and just use `c.flagRootMount` everywhere below and it'll be handled for you, when you call `f.Args()` on line 95 above. :-) 

And then you'll need to update the rest of your `args` usage by one.

**Review by cipherboy — COMMENTED** [review]
Mostly info about how to use optional flags, feel free to ignore if you don't want to :-)

File: command/pki_add_intermediate.go (hunk 2 above)
Comment (divyapola5) [reply command/pki_add_intermediate.go]: Yup, updated the code now. Thanks.

**Review by cipherboy — APPROVED** [review]

**Pull request merged**